benchmark/run.py

```diff
@@ -1,29 +1,133 @@
+                input_size=input_size, output_size=output_size,
+                random_state=42, regualizer=1e-1, learning_rate=1e-2
+            )
+
+            with regression as model:
+                # setup model tester
+                tester = ModelTest(model, dataset, verbose=False, random_state=42)
+                result = tester.test(0.2, 0.1)
+                results_data.append(result)
+        tmp = [regressor_name]
+        tmp.extend(results_data)
+        results.append(tmp)
+
+    # Save results
+    headers = ["Model"]
+    headers.extend(datasets_tex)
+    with open(filename, "w") as text_file:
+        text_file.write(tabulate(results, tablefmt="latex",
+                                          floatfmt = ".3f",
+                                          headers=headers))
+
+def computeTimings(regressors, regressors_tex, datasets,
+                   datasets_tex, filename, no_calls = 50):
+    '''Saves timings for regressors to filename.txt'''
+
+    elapsed_times = []
+    results_times = []
+    for dataset, dataset_name in zip(datasets, datasets_tex):
+        stats = []
+        targets = LabelBinarizer().fit_transform(dataset.target)
+        for regressor, regressor_name in zip(regressors, regressors_tex):
+            # Get sizes
+            input_size = dataset.data.shape[1]
+            output_size = max(dataset.target) + 1
+
+            # intialize model
+            regression = regressor(
+                input_size=input_size, output_size=output_size,
+                random_state=42, regualizer=1e-1, learning_rate=1e-2
+            )
+
+            with regression as model:
+                for _ in range(no_calls):
+                    model.reset()
+                    t = time.perf_counter()
+                    for epoch in range(1000):
+                        model.update(dataset.data, targets)
+                    elapsed_times.append(time.perf_counter() - t)
+                stats.append(computeStats(elapsed_times))
+        tmp = [dataset_name]
+        tmp.extend(stats)
+        results_times.append(tmp)
+
+    headers = ["Dataset"]
+    headers.extend(regressors_tex)
+    with open(filename, 'w') as text_file:
+         text_file.write(tabulate(
+            results_times,
+            tablefmt="latex",
+            headers=headers))
+
+
+
+def main():
+    # Define list of regressors
+    #regressors = [SparseMaxTFPython, SparseMaxTF, SparseMaxPython, SoftMaxTF]
+    regressors = [SparseMaxTF, SoftMaxTF]
+    # Define names to be diplayed in .tex table
+    #regressors_tex = ["TensorFlow Python", "TensorFlow", "Numpy", "SoftMax"]
+    regressors_tex = ["SparseMax","SoftMax"]
+    # Get data
+    digits = datasets.load_digits()
+    iris = datasets.load_iris()
+
+    # Define list of datasets
+    lDataset = [digits, iris]
+
+    # Define names of datasets to be displayed in .tex table
+    dataset_tex = ["MNIST", "Iris"]
+
+    computeResults(regressors, regressors_tex, lDataset, dataset_tex,
```

Comment:
It's not important, but it would be nice if you would split this file up in `results.py` and `timings.py`. Then it will be easier to read.